patches/node/enable_crashpad_linux_node_processes.patch

```diff
@@ -8,53 +8,44 @@ to child processes spawned with `ELECTRON_RUN_AS_NODE` which is used
 by the crashpad client to connect with the handler process.
 
 diff --git a/lib/child_process.js b/lib/child_process.js
-index 73c1dc769542865bdf7a2a03c16cef141d3f4b05..4c725a991b803c2bfde58b674df15b328102450b 100644
+index 73c1dc769542865bdf7a2a03c16cef141d3f4b05..35a096dcff6f7072391bccd7952da7e8ea8980bb 100644
 --- a/lib/child_process.js
 +++ b/lib/child_process.js
-@@ -159,7 +159,6 @@ function fork(modulePath, args = [], options) {
+@@ -59,6 +59,7 @@ let debug = require('internal/util/debuglog').debuglog(
+ );
+ const { Buffer } = require('buffer');
+ const { Pipe, constants: PipeConstants } = internalBinding('pipe_wrap');
++const { getCrashdumpSignalFD, getCrashpadHandlerPID } = process._linkedBinding('electron_common_crashpad_support');
+ 
+ const {
+   AbortError,
+@@ -159,7 +160,6 @@ function fork(modulePath, args = [], options) {
        ArrayPrototypeSplice(execArgv, index - 1, 2);
      }
    }
 -
    args = [...execArgv, modulePath, ...args];
  
    if (typeof options.stdio === 'string') {
-@@ -613,6 +612,22 @@ function normalizeSpawnArguments(file, args, options) {
+@@ -613,6 +613,21 @@ function normalizeSpawnArguments(file, args, options) {
                      'options.windowsVerbatimArguments');
    }
  
 +  if (process.platform === 'linux') {
 +    if (ObjectPrototypeHasOwnProperty(options.env || {}, 'ELECTRON_RUN_AS_NODE') &&
 +        (file === process.execPath)) {
-+      // On Linux, pass the file descriptor which crashpad server process
-+      // uses to monitor the child process.
-+      const fd = process.getCrashdumpSignalFD();
-+      if (fd !== -1) {
++      // On Linux, pass the file descriptor which crashpad handler process
++      // uses to monitor the child process and PID of the handler process.
++      // https://source.chromium.org/chromium/chromium/src/+/110.0.5415.0:components/crash/core/app/crashpad_linux.cc;l=199-206
++      const fd = getCrashdumpSignalFD();
++      const pid = getCrashpadHandlerPID();
++      if (fd !== -1 && pid !== -1) {
 +        options.env.CRASHDUMP_SIGNAL_FD = fd;
-+        // On Linux, pass the PID of the crashpad handler process to the child process
-+        // which is needed by the crashpad client running in the child process.
-+        // https://source.chromium.org/chromium/chromium/src/+/110.0.5415.0:components/crash/core/app/crashpad_linux.cc;l=199-206
-+        ArrayPrototypePush(args, `--crashpad-handler-pid=${process.getCrashpadHandlerPID()}`);
++        options.env.CRASHPAD_HANDLER_PID = pid;
```

Comment:
if options / options.env can be null, do we need to make sure we initialize those as objects here?

Reply:
Yeah, I think that's a good callout, I'll add that in 🙇‍♀️

Reply:
Addressed in [ecd98e8](https://github.com/electron/electron/pull/36460/commits/ecd98e89805fb167b2074a91f3a49e0c4aa68c5a)

Reply:
As a heads up, I reverted this in [9d3bedb](https://github.com/electron/electron/pull/36460/commits/9d3bedb469b57443ed27380d83a55f4efa1d8edf) after seeing multiple node test failures. It looks like node expects the options object to not be extendable here, so we can't create an options.env at this point if it doesn't already exist.

We have a check in node_main that won't initialize crashpad if the fd and pid don't exist, so I think it's safe to not initialize them here, and it seems to be what Node expects.